cmd/login.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"github.com/redhat-developer/odo/pkg/auth"
+	"github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/spf13/cobra"
+)
+
+var (
+	userName string
+	password string
+	token    string
+	caAuth   string
+	skipTLS  bool
+)
+
+// versionCmd represents the version command
+var loginCmd = &cobra.Command{
+	Use:   "login",
+	Short: "Login to cluster",
+	Long:  "Login to cluster",
+	Example: `
+  # Log in interactively
+  odo login
+
+  # Log in to the given server with the given certificate authority file
+  odo login localhost:8443 --certificate-authority=/path/to/cert.crt
+
+  # Log in to the given server with the given credentials (basic auth)
+  odo login localhost:8443 --username=myuser --password=mypass
+
+  # Log in to the given server with the given credentials (token)
+  odo login localhost:8443 --token=xxxxxxxxxxxxxxxxxxxxxxx
+	`,
+	Args: cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var server string
+		if len(args) == 1 {
+			server = args[0]
+		}
+		err := auth.Login(server, userName, password, token, caAuth, skipTLS)
+		if err != nil {
+			util.CheckError(err, "")
+		}
+	},
+}
+
+func init() {
+	// Add a defined annotation in order to appear in the help menu
+	loginCmd.Annotations = map[string]string{"command": "utility"}
+	loginCmd.SetUsageTemplate(cmdUsageTemplate)
+	loginCmd.Flags().StringVar(&userName, "username", userName, "username, will prompt if not provided")
+	loginCmd.Flags().StringVar(&password, "password", password, "password, will prompt if not provided")
+	loginCmd.Flags().StringVarP(&token, "token", "t", token, "token, will prompt if not provided")
```

Comment:
Do we need -t? Because -u and -p are missing

Reply:
`oc login` has it exactly the other way. It has  `-u` and `-p` but doesn't have `-t`.
We should follow the same path, so `odo login` is fully compatible with `oc login`.